build-support/bin/packages.py

```diff
@@ -205,29 +220,55 @@ def get_pgp_program_name() -> str:
     return configured_name or "gpg"
 
 
+@contextmanager
+def set_pants_version(version: str) -> Iterator[None]:
+    """Temporarily rewrite the VERSION file."""
+    original_content = CONSTANTS.pants_version_file.read_text()
+    CONSTANTS.pants_version_file.write_text(version)
+    try:
+        yield
+    finally:
+        CONSTANTS.pants_version_file.write_text(original_content)
+
+
 # -----------------------------------------------------------------------------------------------
 # Script commands
 # -----------------------------------------------------------------------------------------------
 
 
-def build_and_print_packages(version: str) -> None:
-    packages_by_flags = defaultdict(list)
-    for package in sorted(all_packages()):
-        packages_by_flags[package.bdist_wheel_flags].append(package)
+def build_pants_wheels() -> None:
+    banner("Building Pants wheels")
+    version = CONSTANTS.pants_unstable_version
 
-    for flags, packages in packages_by_flags.items():
-        bdist_flags = " ".join(flags)
+    if CONSTANTS.deploy_pants_wheel_dir.exists():
+        shutil.rmtree(CONSTANTS.deploy_pants_wheel_dir)
+    destination = CONSTANTS.deploy_pants_wheel_dir / version
+    destination.mkdir(parents=True)
+
+    def build(packages: Iterable[Package], bdist_wheel_flags: Iterable[str]) -> None:
+        formatted_flags = " ".join(bdist_wheel_flags)
         args = (
             "./v2",
             "setup-py2",
-            f"--args=bdist_wheel {bdist_flags}",
+            f"--args=bdist_wheel {formatted_flags}",
             *(package.target for package in packages),
         )
         try:
-            # We print stdout to stderr because release.sh is expecting stdout to only be package names.
-            subprocess.run(args, stdout=sys.stderr, check=True)
+            subprocess.run(args, check=True)
             for package in packages:
-                print(package.name)
+                found_wheels = sorted(Path("dist").glob(f"{package}-{version}-*.whl"))
+                # NB: For any platform-specific wheels, like pantsbuild.pants, we assume that the
+                # top-level `dist` will only have wheels built for the current platform. This
+                # should be safe because it is not possible to build native wheels for another
+                # platform.
```

Comment:
We were already assuming this. Only formalizing it here.

In a followup, I'll change this so that we can accommodate native wheels (`cp36m` et al).